app/controllers/responsiblepeople/PositionWithinBusinessStartDateController.scala

```diff
@@ -0,0 +1,124 @@
+
+
+  def get(index: Int, edit: Boolean = false, flow: Option[String] = None) = Authorised.async {
+      implicit authContext =>
+        implicit request =>
+          dataCacheConnector.fetchAll map { optionalCache =>
+            (optionalCache map { cache =>
+
+              val bt = ControllerHelper.getBusinessType(cache.getEntry[BusinessMatching](BusinessMatching.key))
+                .getOrElse(BusinessType.SoleProprietor)
+
+              val data = cache.getEntry[Seq[ResponsiblePerson]](ResponsiblePerson.key)
+
+              getResponsiblePersonFromData(data,index) match {
+                case Some(rp@ResponsiblePerson(Some(personName),_,_,_,_,_,_,_,_,_, Some(Positions(positions, Some(startDate))),_,_,_,_,_,_,_,_,_,_,_))
+                  => Ok(position_within_business_start_date(Form2[PositionStartDate](startDate), edit, index, bt,  personName.titleName, positions, displayNominatedOfficer(rp, hasNominatedOfficer(data)), flow))
+                case Some(rp@ResponsiblePerson(Some(personName),_,_,_,_,_,_,_,_,_, Some(Positions(positions, None)),_,_,_,_,_,_,_,_,_,_,_))
+                  => Ok(position_within_business_start_date(EmptyForm, edit, index, bt, personName.titleName, positions, displayNominatedOfficer(rp, hasNominatedOfficer(data)), flow))
+                case _
+                  => NotFound(notFoundView)
+              }
+            }).getOrElse(NotFound(notFoundView))
+          }
+    }
+
+  def post(index: Int, edit: Boolean = false, flow: Option[String] = None) = Authorised.async {
+    implicit authContext =>
+      implicit request =>
+        Form2[PositionStartDate](request.body) match {
+          case f: InvalidForm =>
+            dataCacheConnector.fetchAll map { optionalCache =>
+              (optionalCache map { cache =>
+
+                val bt = ControllerHelper.getBusinessType(cache.getEntry[BusinessMatching](BusinessMatching.key))
+                  .getOrElse(BusinessType.SoleProprietor)
+
+                val data = cache.getEntry[Seq[ResponsiblePerson]](ResponsiblePerson.key)
+
+                getResponsiblePersonFromData(data,index) match {
+                  case s@Some(rp@ResponsiblePerson(Some(personName),_,_,_,_,_,_,_,_,_, Some(Positions(positions,_)),_,_,_,_,_,_,_,_,_,_,_)) =>
+                    BadRequest(position_within_business_start_date(f, edit, index, bt, ControllerHelper.rpTitleName(s), positions, displayNominatedOfficer(rp, hasNominatedOfficer(data)), flow))
+                  case _
+                    => NotFound(notFoundView)
+                }
+              }).getOrElse(NotFound(notFoundView))
+            }
+          case ValidForm(_, data) => {
+            for {
+              _ <- updateDataStrict[ResponsiblePerson](index) { rp => rp.positions match {
+                  case Some(x) => rp.positions(Positions.update(x, data))
+                  case None => throw new IllegalStateException("Positions does not exist, cannot update start date")
```

Comment:
was wondering, whats going to happen to the app after the exception is thrown

Reply:
I've added a logging statement as discussed. The exception is for completeness - the thinking behind it is that it's a state that shouldn't ever occur. To avoid calling get() on an Option, or routing to "Page not found" (which is done frequently elsewhere and should be removed, but that's another issue....), I thought it would be better to throw a descriptive exception and allow Play's error handling to route the user appropriately.